src/routes/chains/handlers.rs

```diff
@@ -33,3 +41,16 @@ pub async fn get_single_chain(
     let info_provider = DefaultInfoProvider::new(&chain_id, &context);
     Ok(info_provider.chain_info().await?.into())
 }
+
+fn map_link(context: &RequestContext, original_link: Option<String>) -> Option<String> {
+    original_link.as_ref().map(|link| {
+        let cursor =
+            PageMetadata::from_cursor(link.split("?").collect::<Vec<&str>>().get(1).unwrap_or(&""))
+                .to_url_string();
```

Comment:
nitpick: For this we can also rely on `Url::parse` and then use `query()` to return the query parameter part (suggesting this because parsing URLs can be complex and this might provide extra validations)